boto/gs/bucket.py

```diff
@@ -453,8 +457,8 @@ def _set_acl_helper(self, acl_or_str, key_name, headers, query_args,
             headers['x-goog-if-metageneration-match'] = str(if_metageneration)
 
         response = self.connection.make_request(
-            'PUT', get_utf8_value(self.name), get_utf8_value(key_name),
-            data=get_utf8_value(data), headers=headers, query_args=query_args)
+            'PUT', self.name, key_name,
```

Comment:
The HTTP request should probably accept bytes (str in Python2, bytes in Python3).  I see two options here:

- If we're going to remove this wrapper from all the call sites, I'd propose invoking `get_utf8_value` on each of the arguments within the method body of `make_connection` in connection.py.  Note that in the event we need to send a set of bytes encoded in some other encoding, this would be fine -- `get_utf8_value` only ensures that the type is bytes, not that they were encoded in UTF8 coding.

- Or, go back to requiring that each caller passes bytes (not unicode) to `make_request`.

Reply:
Actually, after talking this over with Ed offline, it looks like the s3 components removed these utf8 wrappers from their `make_connection` calls.

Reply:
Ah, I see. The get_utf8_value wrappers were added ~5 years ago [1], but the code was refactored roughly a year later [2] to prevent the issues from mixed unicode/bytes described in [1].

[1] https://github.com/boto/boto/commit/8f466ebcac5fa22743bae37ef3d3312c79348a9b
[2] https://github.com/boto/boto/commit/96cd2800b6db067e503ef499486adf50ee9ca928